### PR TITLE
Homework14: Marie Ambrazhei 

### PR DESCRIPTION
**Pull request by MarieAmbrazhei**

### CODE REVIEW

File: homework_14/homework_14_2.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                          '4 for division\n'
+                          '5 for exponentiation\n'
+                          '6 to input a whole expression\n'))
+                if 1 <= self.menu_number <= 6:
+                    return self.menu_number
+                print('You might choose only 1 to 6')
+            except ValueError:
+                print('Invalid input. Please enter a number between 1 and 6.')
+
+    def expression(self, input_expr):
```
Comment (Stml89) [inline homework_14/homework_14_2.py:34]: not bad

File: homework_14/homework_14_2.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+calculator = SimpleCalculator()
+calculator.get_menu_number()
+calculator.calc()
+assert calculator.summary(Decimal('3'), Decimal('2')) == '3 + 2 = 5'
+assert calculator.subtraction(Decimal('5'), Decimal('2')) == '5 - 2 = 3'
+assert calculator.multiplication(Decimal('3'), Decimal('2')) == '3 * 2 = 6'
+assert calculator.division(Decimal('3'), Decimal('2')) == '3 / 2 = 1.5'
+assert calculator.division(Decimal('5'),
+                           Decimal('0')) == 'Error! Division by zero!'
+assert calculator.exponentiation(Decimal('2'), Decimal('3')) == '2 ^ 3 = 8'
```
Comment (Stml89) [inline homework_14/homework_14_2.py:123]: Internally you are using it as a number, so you can use it here as well
```suggestion
assert calculator.summary(3, 2) == '3 + 2 = 5'
assert calculator.subtraction(5, 2) == '5 - 2 = 3'
assert calculator.multiplication(3, 2) == '3 * 2 = 6'
assert calculator.division(3, 2) == '3 / 2 = 1.5'
assert calculator.division(5, 0) == 'Error! Division by zero!'
assert calculator.exponentiation(2, 3) == '2 ^ 3 = 8'
```

Comment (Stml89) [reply homework_14/homework_14_2.py:123]: You have implemented tests for all internal methods, which is really nice!

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework_14/homework_14_1.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+def add_result_info(file_path, info):
+    """Append the result information to the file."""
+    if os.path.exists(file_path):
+        with open(file_path, 'a', encoding='utf-8') as file:
+            file.write(info)
+    else:
+        print(f'File {file_path} does not exist')
+
+
+add_result_info(file_path='student.txt', info=entire_info)
```
[the remark this answers is not in the document]
Comment (MarieAmbrazhei) [reply homework_14/homework_14_1.py:71]: fixed

**Review by Stml89 — APPROVED** [review]

**Pull request merged**